adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m

```diff
@@ -66,38 +61,60 @@ - (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguratio
   }
 
   NSError *error = nil;
-  _bannerSize = [self bannerSizeFormGADAdSize:adConfiguration.adSize error:&error];
+  PAGBannerAdSize bannerSize = [self bannerSizeFormGADAdSize:adConfiguration.adSize error:&error];
   if (error) {
     _loadCompletionHandler(nil, error);
     return;
   }
-  _nativeExpressBannerView =
-      [[BUNativeExpressBannerView alloc] initWithSlotID:placementId
-                                     rootViewController:adConfiguration.topViewController
-                                                 adSize:_bannerSize];
-  _nativeExpressBannerView.delegate = self;
-  [_nativeExpressBannerView setAdMarkup:adConfiguration.bidResponse];
+  
+  PAGBannerRequest *request = [PAGBannerRequest requestWithBannerSize:bannerSize];
+  request.adString = adConfiguration.bidResponse;
+  
+  __weak typeof(self) weakSelf = self;
+  [PAGBannerAd loadAdWithSlotID:placementId
+                        request:request
+              completionHandler:^(PAGBannerAd * _Nullable bannerAd,
+                                  NSError * _Nullable loadError) {
+    __strong typeof(weakSelf) strongSelf = weakSelf;
+    
+    if (loadError) {
+      strongSelf->_loadCompletionHandler(nil,loadError);
+      return;
+    }
+    
+    CGRect frame = bannerAd.bannerView.frame;
+    frame.size = bannerSize.size;
+    bannerAd.bannerView.frame = frame;
+    bannerAd.rootViewController = adConfiguration.topViewController;
+    
+    strongSelf->_bannerAd = bannerAd;
+    strongSelf->_bannerAd.delegate = strongSelf;
+    
+    if (strongSelf->_loadCompletionHandler) {
+      strongSelf->_delegate = strongSelf->_loadCompletionHandler(strongSelf,nil);
+    }
```

Comment:
Question: What is the reason that we need `strong` and `weak` here? What leads to this difference in the major release?

I don't see such implementation in Pangle documentation: https://www.pangleglobal.com/integration/ios-banner-ads.

Reply:
Seems like it's because of these lines:
```
    if (strongSelf->_loadCompletionHandler) {
      strongSelf->_delegate = strongSelf->_loadCompletionHandler(strongSelf,nil);
    }
```
But still it would be great if you can clarify that in Pangle SDK perspective.

Reply:
To prevent circular references